src/html.rs

```diff
@@ -364,7 +364,7 @@ fn gen_html(
                 .map(|&v| v as i64)
                 .unwrap_or(-1);
 
-            (index, count, l.unwrap())
+            (index, count, String::from_utf8_lossy(&l.unwrap()).into_owned())
```

Comment:
If I understand correctly, `from_utf8_lossy` is applied to each line of text. Why not apply it one time to all the text ?
Generally speaking, we should probably either detect the encoding or add an option to specify it and then use the correct decoder.

Reply:
About detecting encoding, I wrote this code few years ago:
https://github.com/mozilla/rust-parsepatch/blob/b3dbb48ecc73de3fecea20c7e8848a3fc2239772/tests/test.rs#L188-L201

Reply:
I kept it on a per-line basis to avoid allocating a buffer for the whole file and keep the streaming nature of it, but certainly could change it to happen per-file.

Encoding detection wise, I could add a similar sort of detection, it was not necessary for my case, so I didn't think about it.